### ISSUE TITLE
[Feature Request] Mark as watched only when x% is actually watched

### ISSUE THREAD

**Issue by trivarialthea** [aliyss/syncribullet#6]
Instead of marking a movie or episode of a TV show as watched instantly (as is done now), only mark it when a certain threshold is reached or when it is marked as "watched" in streamio.

The current implementation is frustrating because, for instance, if you have to leave in the middle of an episode and turn off your PC, it will already be marked as watched on SIMKL. Which can confuse someone in the long term.

Would it be possible?

**Comment by aliyss** [comment]
Right. So what I am doing right now is abusing the subtitle handler to check if a stream has been started.

Stremio provides no way to check start or end of an episode.

I have been playing around with the idea to access the stremio user library. So if it has been marked as watched by stremio, then it will go and mark it as watched on simkl. But this has it's own issues with the website basically needing a timer to recheck if an episode has been watched.

**Comment by trivarialthea** [comment]
Ah, so streamio does not directly support tracking this? How does the built-in trakt connection/addon work, then, is that like an exception that was made directly from streamio?

**Comment by aliyss** [comment]
You know I had a look at some trakt addon stuff... but as far as I know there is no direct support to what we are trying to achieve. The other stuff is most likely built in, because I saw some trakt stuff within stremio-core

**Comment by aliyss** [comment]
Ok, so I am working on some validation option. Basically you will have to then authenticate with Stremio. And the addon can then read your library.

I plan on doing it like this:

- You start watching.
- It triggers the Subtitle Handler
- Server recognizes what is being watched and gets the runtime of the movie/episode
- Server waits that amount of time + say 5-10mins
- Server uses your Stremio authentication to check if the movie/episode has been marked as watched on Stremio
- If true -> Mark as watched
- If false -> Don't do anything

Now there is one issue, I would like to discuss... I'll anyways add an option for this, but want to discuss the default setting.

If the stremio auth fails for some reason or needs to be regenerated: Do you want it to not do anything or would you prefer it to sync on start again?

**Comment by trivarialthea** [comment]
I suppose if we take the BFU mentality of "Set it and forget it" the 2nd option where it would automatically sync again makes more sense. 

The problem is, fresh install of streamio starts with nothing watched, therefore on initial install (if you want to achieve 2 way sync) is to sync SIMKL into streamio first. Or we could leave it as one way and only sync from streamio to SIMKL. 

If not, there is a possibility that it would wipe the entire SIMKL list.

At least I think so, in theory...

**Comment by aliyss** [comment]
Ok... So to clarify, my add-on does not sync complete lists... It only syncs newly watched stuff. So there shouldn't be an issue either way when reinstalling stremio or starting new. Nothing is ever deleted, only added.

**Comment by trivarialthea** [comment]
Perfect

**Comment by husoyo** [comment]
Any update on this?

**Comment by aliyss** [comment]
This is still in discussion with the stremio team. I decided for now not to prioritize this on request from stremio, because they don't want me to "abuse" the internal api.

Depending on how the pacing of the implementation is, I might/might not continue on my own. For now I'll respect it afaican

**Comment by ChakraFusion** [comment]
I came here for a similar and mostly related issue.
First if all I really appreciate the work you're putting into this. I recently came from trakt.tv and switched to simkl because trakt limited the watchlist entry amount.
The problem with the sync on start for me is that oftentimes I look into some movies for a couple minutes and then not watch them. Of course that not good together with the sync on start.

A 2nd issue I experienced at least with 3 separate series until now is that complete series with all seasons get marked as watched after I watched any random episode. It happened with 'the penguin' and with 'my hero academia' for example. Maybe that is caused by 'complete series'-style or 'complete season'-style uploads in Stremio / RD.

**Comment by erintesden** [comment]
So... In seeing it's almost a year since this, has it been any possible solution/implemention to this? I

**Comment by ThePotatoBomb** [comment]
Im also hoping there's some progress on this
